mesheryctl/pkg/utils/helpers_test.go

```diff
@@ -0,0 +1,498 @@
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+func getFixturesDirectory() string {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		log.Fatal("TestBackupConfigFile: Cannot get current working directory")
+	}
+	currDir := filepath.Dir(filename)
+	// get the fixtures file directory
+	fixturesDir := filepath.Join(currDir, "fixtures")
+	return fixturesDir
+}
+
+var fixturesDir = getFixturesDirectory()
+
+func TestBackupConfigFile(t *testing.T) {
+	name := "config.yaml"
+	configFilePath := filepath.Join(fixturesDir, name)
+
+	backupFileName := "config.bak.yaml"
+	backupConfigFilePath := filepath.Join(fixturesDir, backupFileName)
+
+	// creates a config file
+	NewGoldenFile(t, name, fixturesDir).Write("mesheryctl")
+
+	BackupConfigFile(configFilePath)
+
+	// check if backup file is present or not
+	_, err := os.Stat(backupConfigFilePath)
+	if err != nil {
+		t.Errorf("BackupConfigFile error = %v", err)
+	}
+}
+
+func TestStringWithCharset(t *testing.T) {
+	// checking the length, since this function returns random strings everytime
+	strLength := 10
+
+	gotString := StringWithCharset(strLength)
+
+	if len(gotString) != strLength {
+		t.Errorf("StringWithCharset got = %v want = %v", len(gotString), strLength)
+	}
+}
+
+func TestPrereq(t *testing.T) {
+	_, _, err := prereq()
+	if err != nil {
+		t.Errorf("prereq error = %v", err)
+	}
+}
+
+func TestSetFileLocation(t *testing.T) {
+	err := SetFileLocation()
+	if err != nil {
+		t.Errorf("SetFileLocation error = %v", err)
+	}
+}
```

Comment:
Took me some time to figure out the problem. It's basically this function. @piyushsingariya